torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -523,11 +541,14 @@ struct GraphFuser {
   std::pair<graph_node_list::iterator, bool> scanNode(Node * consumer) {
     auto stage_guard = block->owningGraph()->setStageTemporary(consumer->stage());
     if(isFusableAsExitNode(consumer)) {
+      value_list inputs;
+      auto consumer_inputs = consumer->kind() == aten::cat ?
```

Comment:
This is a bit awkward. One way to simplify the logic would be to first apply a fuse-cat pass, and then do the graph fuser. But that would require either implementing FusedCat as an op or implementing a fission cat pass. I think this is fine for now. But if we add more special case (e.g. split/chunk handling) we might want to formalize the "create special versions we care a bout", do fusion, "revert special versions" path.

Reply:
I agree. I just didn't want to temporarily fuse and later unfuse cats as you mentioned. The situation is not too bad at the moment, so I just decided to go with this, but I agree that if we'll want to make the fuser more powerful this will need a refactor/